scripts/dts/edtlib.py

```diff
@@ -1390,9 +1390,9 @@ class Binding:
       The free-form description of the binding.
 
     compatible:
-      The compatible string the binding matches. This may be None if the
-      Binding object is a child binding or if it's inferred from node
-      properties.
+      The compatible string the binding matches. If the Binding object
+      is parsed from a child binding, this may be the 'compatible' value of the
```

Comment:
"...from a child binding *that does not have a `compatible` property*, this may be..."

Or something that makes it clear that if the child has a compatible it does not *have* to be equal to the parent.

Reply:
> Or something that makes it clear that if the child has a compatible it does not _have_ to be equal to the parent.

Sure, I've tried to reword this to make that more clear. I added a test as well.

Reply:
I think my problem is with "if condition, then this may happen", when I'd like to see more clearly what *will* happen in different conditions.